typescript/sdk/src/providers/MultiProvider.ts

```diff
@@ -289,9 +289,22 @@ export class MultiProvider<MetaExt = {}> extends ChainMetadataManager<MetaExt> {
     factory: F,
     params: Parameters<F['deploy']>,
   ): Promise<Awaited<ReturnType<F['deploy']>>> {
+    // setup contract factory
     const overrides = this.getTransactionOverrides(chainNameOrId);
     const signer = this.getSigner(chainNameOrId);
-    const contract = await factory.connect(signer).deploy(...params, overrides);
+    const contractFactory = await factory.connect(signer);
+
+    // estimate gas
+    const deployTx = contractFactory.getDeployTransaction(...params, overrides);
+    const gasEstimated = await signer.estimateGas(deployTx);
+
+    // deploy with 10% buffer on gas limit
+    const contract = await contractFactory.deploy(...params, {
+      ...overrides,
+      gasLimit: gasEstimated.add(gasEstimated.div(10)), // 10% buffer
+    });
+
```

Comment:
curious when the gas limit of a CREATE tx would change?

Reply:
possible from RPC problems leading to misestimation? I've encountered issues on polygon, bsc, goerli and a few others where the tx fails because the gas limit automatically set is _just_ lower than required